instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -40,6 +46,49 @@ def process(commands) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
 
           private
 
+          # Ensure all values are properly encoded and truncated before
+          def format_commands(value)
+            return nil if value.nil?
+            return value.map { |v| format_commands(v) } if value.is_a?(Array)
+
+            value = OpenTelemetry::Common::Utilities.utf8_encode(value, binary: true)
+            OpenTelemetry::Common::Utilities.truncate(value, MAX_VALUE_LENGTH)
+          end
+
+          # Examples of commands received for parsing
+          # Redis#queue     [[["set", "v1", "0"]], [["incr", "v1"]], [["get", "v1"]]]
+          # Redis#pipeline: [["set", "v1", "0"], ["incr", "v1"], ["get", "v1"]]
+          # Redis#hmset     [["hmset", "hash", "f1", "1234567890.0987654"]]
+          # Redis#set       [["set", "K", "0"]]
+          def parse_commands(commands)
+            commands.map do |command|
+              # We are checking for the use of Redis#queue command, if we detect the
+              # extra level of array nesting we return the first element so it
+              # can be parsed.
+              command = command[0] if command.is_a?(Array) && command[0].is_a?(Array)
+
+              # If we receive an authentication request command
+              # we want to short circuit parsing the commands
+              # and return the obfuscated command
+              return 'AUTH ?' if command[0] == 'auth'
+
+              cmd = command[0].upcase!
+              if config[:enable_statement_obfuscation]
+                cmd + ' ?' * command[1..-1].size
```

Comment:
`cmd` will be `nil` here if `upcase!` made no changes.

There's no value in slicing `command` just to compute the size - you can just `' ?' * (command.size - 1)`

Reply:
Good catch, I didn't even consider that you could do something like this 👇 
```rb
      redis.queue(['SET', 'v1', '0'])
      redis.commit
```

I was under the impression that the commands always came in as symbols.